olive/auto_optimizer/regulate_mixins.py

```diff
@@ -104,19 +104,21 @@ def _regulate_fp16(self, pass_config, pass_flows):
         return pass_config, pass_flows
 
     def regulate_data_config(self, pass_config, pass_flows):
-        from olive.workflows.run.config import INPUT_MODEL_DATA_CONFIG
-
-        data_configs = self.data_configs.get(INPUT_MODEL_DATA_CONFIG)
-        if not data_configs:
+        if not self.auto_optimizer_config or self.auto_optimizer_config.disable_auto_optimizer:
             return pass_config, pass_flows
 
-        # data_config
-        passes_require_data_config = ["OnnxQuantization", "OrtPerfTuning"]
+        if len(self.data_configs) != 1:
+            raise ValueError("AutoOptimizer expects exactly one data config.")
+
+        data_config = self.data_configs[0].to_json()
```

Comment:
nit: this is not necessary since the pass config can accept a data config object.

Reply:
Done!